Source/quests.cpp

```diff
@@ -829,4 +816,18 @@ void SetMultiQuest(int q, quest_state s, bool log, int v1)
 	}
 }
 
+bool QuestStruct::IsAvailable()
+{
+	if (setlevel)
+		return false;
+	if (currlevel != _qlevel)
+		return false;
+	if (_qactive == QUEST_NOTAVAIL)
+		return false;
+	if (gbIsMultiplayer && QuestData[_qidx].isSinglePlayerOnly)
+		return false;

```

Comment:
Reading over @ephphatha comment again I see that he also pointed out that this part could be determined when quests are initialized (or loaded), as a new parameter, which would allow IsAvailable() to be simplified a bit and moved to the header as a `cosnt` function.

This would be a decent improvement to do at a later point.